WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -571,4 +571,101 @@
+    let!(:registration2) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: registered_user) }
+    let!(:registration3) { FactoryGirl.create(:registration, competitionId: past_competition1.id, user: registered_user) }
+    let!(:registration4) { FactoryGirl.create(:registration, competitionId: past_competition3.id, user: organizer) }
+    let!(:registration5) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: delegate) }
+    let!(:results_person) { FactoryGirl.create(:person, id: "2014PLUM01", name: "Jeff Plumb") }
+    let!(:results_user) { FactoryGirl.create :user, name: "Jeff Plumb", wca_id: "2014PLUM01" }
+
+    context 'when not signed in' do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :my_competitions
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as user with results for a comp they did not register for' do
+      before do
+        sign_in results_user
+      end
+
+      it 'shows my upcoming and past competitions' do
+        Result.create!(
+          pos: 1,
+          personId: "2014PLUM01",
+          personName: "Jeff Plumb",
+          countryId: "AUS",
+          competitionId: past_competition1.id,
+          eventId: "333",
+          roundId: "f",
+          formatId: "a",
+          value1: 2000,
+          value2: 2000,
+          value3: 2000,
+          value4: 2000,
+          value5: 2000,
+          best: 2000,
+          average: 2000,
+          regionalSingleRecord: "WR",
+          regionalAverageRecord: "WR"
+        )
+        get :my_competitions
+        expect(assigns(:not_past_competitions)).to eq []
+        expect(assigns(:past_competitions)).to eq [past_competition1]
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      before do
+        sign_in registered_user
+      end
+
+      it 'shows my upcoming and past competitions' do
+        get :my_competitions
+        expect(assigns(:not_past_competitions)).to eq [future_competition1, future_competition3]
+        expect(assigns(:past_competitions)).to eq [past_competition1]
+      end
+    end
+
+    context 'when signed in as an organizer' do
+      before do
+        sign_in organizer
+      end
+
+      it 'shows my upcoming and past competitions' do
+        get :my_competitions
+        expect(assigns(:not_past_competitions)).to eq [future_competition1, future_competition2, future_competition3]
+        expect(assigns(:past_competitions)).to eq [past_competition1, past_competition3]
+      end
+    end
+
+    context 'when signed in as a delegate' do
+      before do
+        sign_in delegate
+      end
+
+      it 'shows my upcoming and past competitions' do
+        unscheduled_competition1.update_columns(year: 0, month: 0, day: 0, endMonth: 0, endDay: 0)
```

Comment:
And did you try to set both start and end dates to nil in this way? If you specify only the start_date, the validation throws an exception because we compare the dates unless both are nil (I'm not able to check it right now and that's why I cannot play with it).